src/repoSyncResult.ts

```diff
@@ -38,8 +38,12 @@ export class RepoSyncResult {
   }
 
   /** Whether we treat is as still reliable data in absence of a more recent successful sync. */
-  isRecent(): boolean {
-    return this.syncStartUnixMillis >= Date.now() - 1000 * 60 * 5;
+  isRecent(lastSyncDurationMillis: number): boolean {
+    const startIsWithinLast5Minutes =
+      this.syncStartUnixMillis >= Date.now() - 1000 * 60 * 5;
+    const isWithin3SyncsDuration =
```

Comment:
depending on comments settings syncs can take more than 5 minutes for me, so constant "sync staleness" condition stopped working really. Need to be smarter

Reply:
actually never mind, it's missing await